src/endpoint/s3/s3_utils.js

```diff
@@ -622,6 +622,16 @@ function encode_uri_unless_already_encoded(uri = '') {
     return is_uri_already_encoded(uri) ? uri : encodeURIComponent(uri);
 }
 
+/**
+ * version_id_empty_string_check throws an error if version_id is an empty string
+ * @param {string} version_id
+ */
+function version_id_empty_string_check(version_id) {
+    if (version_id?.length === 0) {
```

Comment:
Why did you prefer the length check over explicitly checking if the version ID is an empty string?
```suggestion
    if (version_id === '') {
```

Reply:
oh I agree here too, the explicit empty string check is better than fishing for the length.

Reply:
checking for explicit empty string is better and more readable imho, and I'm even ready to fight with google about it.